DevAcademy/Scenes/PlaceDetail/PlaceDetailViewModel.swift

```diff
@@ -2,24 +2,161 @@ import SwiftUI
+        }
+        return "-"
+    }
+    
+    var webIsAvailable: Bool {
+        if attributeIsFetch(place.attributes.web) {
+            return true
+        }
+        return false
+    }
+    
+    var webPlaceholder: String {
+        simpleStringUrl(from: placeWeb)
     }
     
-    var placeImage: URL {
-        place.properties.obrId1
+    // PHONE
+    var placePhone: String { // place phone return "123456789", "0", "-"
+        if attributeIsFetch(place.attributes.phone) {
+            return place.attributes.phone!.filteringWhiteSpace()
+        }
+        return "-"
+    }
+    
+    var phonePlaceholder: String {
+        formatPhoneNumber(placePhone)
+    }
+    
+    // EMAIL
+    var placeEmail: String {
+        if attributeIsFetch(place.attributes.email) {
+            return place.attributes.email!.filteringWhiteSpace()
+        }
+        return "-"
+    }
+    
+    // PROGRAMME
+    var placeProgramme: String {
+        checkAndFixHasprefix(link: place.attributes.programme!) // can force unwrapp - check value != nil in programmeIsAvailable
+    }
+    
+    var programmeIsAvailable: Bool {
+        if attributeIsFetch(place.attributes.programme) {
+            return true
+        }
+        return false
+    }
+    
+    
+    // MARK: FUNCTIONS
+    func attributeIsFetch(_ attribute: String?) -> Bool {
+        if attribute != nil {
+            return true
+        }
+        return false
     }
     
     func addFavorites() {
         isTappedFavorite.toggle()
     }
-}
+    
+    func checkAndFixHasprefix(link: String) -> String {
+        let httpProtocol = "https://"
+        
+        if link.hasPrefix("http://") || link.hasPrefix("https://") {
+            return link
+        } else {
+            return httpProtocol + link
+        }
+    }
+    
+    func simpleStringUrl(from urlString: String) -> String {
+        if urlString == "-" {
+            return urlString.self
+        } else {
+            let urlComponents = URLComponents(string: urlString)! // --> http//www.example.com
+            let prefixToRemove = "www."
+            let relativeURL = urlComponents.host! // --> www.example.com
+            if relativeURL.hasPrefix(prefixToRemove) {
+                return String(relativeURL.dropFirst(prefixToRemove.count)) // --> example.com
+            }
+            return relativeURL
+        }
+    }
 
+    func formatPhoneNumber(_ number: String) -> String {
+        switch number {
+        case "0":
+            return "-"
+        case "-":
+            return number.self
+        default:
+            let numberToInt = Int(number)
+            let numberFormatter = NumberFormatter()
+            numberFormatter.numberStyle = .decimal
+            numberFormatter.groupingSeparator = " " // --> 123 456 789
+            let formattedNumber = numberFormatter.string(from: NSNumber(value: numberToInt!)) ?? "-"
+            
+            return "+420 " + formattedNumber
```

Comment:
asi bych byl opatrnej s tim pridavanim "+420" na zacatek 🤔 ...mozna bych si overil, jestli uz tam treba neni a nebo klidne nechal telefonni cislo tak jak prijde. Aby se nestalo, ze tam nekdo bude mit dvakrat za sebou +420 nebo ze pridame predcislo cislu, ktere neni ceske.

Reply:
"+420" ve výchozím JSONu není, mám to už dost prozkoumané, ale je pravda, že ne vždy to takhle můžu projít, kdyby těch položek bylo třeba 1000, tak určitě všechny neprojdu. Jelikož se jedná o Brno, předpokládala jsem, že by měli být všechna čísla česká, ale člověk nikdy neví, stejně tak jako jsou různé chyby v datech atd. Ať si to zbytečně nekomplikuju, tak to oddělám, stačí jenom číslo.